File: java/polimorfism_java/Cone.java
public class Cone extends Figura {
    public Cone(String nome, float raioBase, float altura) {
        super(nome,raioBase,altura);
    }
    
    @Override
    public void display() {
        System.out.println("nome: "+getNome());
        System.out.println("raio: "+getDim1());
        System.out.println("altura: "+getDim2());
        System.out.println("area: "+getArea());
        System.out.println("volume: "+getVolume());
    }

    @Override
    public float getArea() {
        float slantHeight = (float)Math.sqrt(Math.pow(getDim1(),2)+Math.pow(getDim2(),2));
        float areaLateral = (float)(Math.PI*getDim1()*slantHeight);
        float areaBase = (float)(Math.PI*Math.pow(getDim1(),2));
        return areaLateral + areaBase;
    }

    @Override
    public float getVolume() {return (float)((1.0/3.0)*Math.PI*Math.pow(getDim1(),2)*getDim2());}
}

File: exercises/polimorfism/Cone.cpp
#include "Cone.h"
#include <iostream>
#include <cmath>
using namespace std;

Cone::Cone(std::string nome, float base, float altura)
    : Figura(nome, base, altura) {}

void Cone::display(){
    cout<<"nome: "<<nome<<endl;
    cout<<"raio: "<<getDim1()<<endl;
    cout<<"altura: "<<getDim2()<<endl;
    cout<<"area: "<<Cone::getArea()<<endl;
    cout<<"volume: "<<Cone::getVolume()<<endl;
}

float Cone::getArea() const{
    float slant_h = sqrt(getDim1()*getDim1() + getDim2()*getDim2());
    float area_l = M_PI*getDim1()*slant_h;
    float area_b = M_PI*getDim1()*getDim1();
    return area_l+area_b;
}
float Cone::getVolume() const{
    return (1.0f/3.0f)*M_PI*getDim1()*getDim1()*getDim2();
}